foolbox/models/jax.py

```diff
@@ -8,7 +8,20 @@
 
 class JAXModel(ModelWithPreprocessing):
     def __init__(
-        self, model: Any, bounds: BoundsInput, preprocessing: Preprocessing = None
+        self,
+        model: Any,
+        bounds: BoundsInput,
+        preprocessing: Preprocessing = None,
+        data_format: Optional[str] = "channels_last",
     ):
         dummy = ep.jax.numpy.zeros(0)
         super().__init__(model, bounds=bounds, dummy=dummy, preprocessing=preprocessing)
+        self._data_format = data_format
+
+    @property
+    def data_format(self) -> str:
+        if self._data_format is None:
+            raise AttributeError(  # AttributeError -> hasattr returns False
```

Comment:
I don't think that's true. Raising an AttributeError inside a function won't change the output of `hasattr` because `hasattr` doesn't execute the function (just tried that).

Reply:
Are you sure @jonasrauber ? The [python documentation](https://docs.python.org/3/library/functions.html#hasattr) for `hasattr` says: _The result is True if the string is the name of one of the object’s attributes, False if not. (This is implemented by calling getattr(object, name) and seeing whether it raises an AttributeError or not.)_

Therefore, I don't understand why `hasattr` shouldn't execute get_attr...

Reply:
True, with the added `@property` it should be fine